core/test/financial_templates/PricelessPositionManager.js

```diff
@@ -591,5 +591,112 @@ contract("PricelessPositionManager", function(accounts) {
+      from: tokenHolder
+    });
+    let settleExpiredResult = await pricelessPositionManager.settleExpired({ from: tokenHolder });
+    const tokenHolderFinalCollateral = await collateral.balanceOf(tokenHolder);
+    const tokenHolderFinalSynthetic = await tokenCurrency.balanceOf(tokenHolder);
+
+    // The token holder should gain the value of their synthetic tokens in underlying.
+    // The value in underlying is the number of tokens they held in the beginning * settlement price as TRV
+    const expectedTokenHolderFinalCollateral = tokenHolderInitialSynthetic.muln(redemptionPrice);
+    assert(tokenHolderFinalCollateral.sub(tokenHolderInitialCollateral), expectedTokenHolderFinalCollateral);
+
+    // The token holder should have no synthetic positions left after settlement.
+    assert(tokenHolderFinalSynthetic, 0);
+
+    //Check the event returned the correct values
+    truffleAssert.eventEmitted(settleExpiredResult, "SettleExpiredPosition", ev => {
+      return (
+        ev.caller == tokenHolder &&
+        ev.collateralAmountReturned == tokenHolderFinalCollateral.sub(tokenHolderInitialCollateral).toString() &&
+        ev.tokensBurned == tokenHolderInitialSynthetic.toString()
+      );
+    });
+
+    // For the sponsor, they are entitled to the underlying value of their remaining synthetic tokens + the excess collateral
+    // in their position at time of settlement - final fees. The sponsor had 100 units of collateral in their position and the final TRV
+    // of their synthetics they sold is 60. Their redeemed amount for this excess collateral is the difference between the two.
+    // The sponsor also has 25 synthetic tokens that they did not sell. This makes their expected redemption = 100 - 60 + 25 * 1.2 - 1 = 69
+    const sponsorInitialCollateral = await collateral.balanceOf(sponsor);
+    const sponsorInitialSynthetic = await tokenCurrency.balanceOf(sponsor);
+
+    // Approve tokens to be moved by the contract and execute the settlement.
+    await tokenCurrency.approve(pricelessPositionManager.address, sponsorInitialSynthetic, {
+      from: sponsor
+    });
+    await pricelessPositionManager.settleExpired({ from: sponsor });
+    const sponsorFinalCollateral = await collateral.balanceOf(sponsor);
+    const sponsorFinalSynthetic = await tokenCurrency.balanceOf(sponsor);
+
+    // The token Sponsor should gain the value of their synthetics in underlying
+    // + their excess collateral from the over collateralization in their position
+    const expectedSponsorCollateralUnderlying = toBN(amountCollateral)
+      .sub(toBN(numTokens).muln(redemptionPrice))
+      .sub(toBN(finalFeePaid));
+    const expectedSponsorCollateralSynthetic = sponsorInitialSynthetic.muln(redemptionPrice);
+    const totalSponsorCollateralReturned = expectedSponsorCollateralUnderlying + expectedSponsorCollateralSynthetic;
+    assert(sponsorFinalCollateral.sub(sponsorInitialCollateral), totalSponsorCollateralReturned);
```

Comment:
nit: why `assert` vs `assert.equal`?

Here and below.

Reply:
no reason, will modify. In general do we prefer to use assert.isEqual versus assert?

Reply:
ah, I did this because `assert` is used instead of `assert.equal` in the "Redemption Post expiry" test. I'll edit those tests too

Reply:
yeah weirdly this test does not pass if you replace with `assert.equal` because it tries to compare bignumbers. I tried to `.eq` the two arguments but it doesn't return the exact same big number

Reply:
I believe it is a rounding error that is somehow ignored with `assert` these are the two arguments being compared when printed out individually `toString()`:
- `tokenHolderFinalCollateral.sub(tokenHolderInitialCollateral).toString() = 60000000000000000000`
- `expectedTokenHolderFinalCollateral.toString() = 59998198560095364710`

Reply:
The only time we normally use `assert` is when we are testing `didContractThrow`, and that's only because it returns something *truthy*, but not true or false.

Reply:
If there's a rounding error, we should make that part of what we test for and understand why it's there.

Reply:
ok I resolved this, it was an error with the `bn.js` `muln` method that doesn't correctly multiply `50000000000000000000` * `1.2`, so i hardcoded the expected answers instead